lib/publiccloud/instance.pm

```diff
@@ -349,6 +349,41 @@ sub wait_for_guestregister {
     die('guestregister didn\'t end in expected timeout=' . $args{timeout});
 }
 
+=head2 update_instance_ip
+
+    update_instance_ip(timeout => 600)
+
+This subroutine checks the public IP cloud provider provides for the VM.
+When the IP differs from `$self->public_ip` we update `$self->public_ip`.
+
+=cut
+
+sub update_instance_ip {
+    my ($self, %args) = @_;
+    $args{timeout} //= 600;
+    my $delay = $args{timeout} > 180 ? 5 : 1;
+
+    # skip SLES4SAP as incompatible with get_public_ip
+    if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        my $public_ip_from_provider = $self->provider->get_public_ip();
+        
+        my $duration;
+        my $start_time = time();
+        until ($public_ip_from_provider !~ /null/ || ($duration = time() - $start_time) >= $args{timeout}) {
+            sleep($delay);
+            $public_ip_from_provider = $self->provider->get_public_ip();
+        }
+
+        # Update the public IP address if it differs
+        if ($self->public_ip ne $public_ip_from_provider) {
+            record_info('IP CHANGED', "The address we know is $self->{public_ip} but provider returns $public_ip_from_provider", result => 'fail');
+            $self->public_ip($public_ip_from_provider);
+        }
+    }
+
+    return true;
```

Comment:
```suggestion
    return $changed;
```

Reply:
I don't need to know if it `$changed`.

Reply:
well, I think it would add value to the routine; but if you don't want it then neither need to retunr `true`:
`return;`

Reply:
To me it is ok the external call, but complex; so eventually, as I wrote [here](https://suse.slack.com/archives/C02CGKBCGT1/p1756380328813069?thread_ts=1756293271.534829&cid=C02CGKBCGT1), to simplify the maintenance, the new routine could be called **here** directly, mantaining its objects indipendency, but avoiding to add it before every `wait_for_ssh` call. Up to you 👍🏼